packages/plugin-typedoc/src/index.ts

```diff
@@ -16,10 +16,15 @@ export interface PluginTypeDocOptions {
    * @default 'api'
    */
   outDir?: string;
+  /**
+   * A function to setup the TypeDoc Application.
+   * @param app
+   */
+  setup?: (app: Application) => Promise<void> | void;
```

Comment:
```suggestion
  typedoc?: (app: Application) => Promise<Application> | Promise<void> | void;
```

I think such an API would be more flexible if users want to replace the already generated typedoc instance in Rspress just like `tools.rspack`

- https://rsbuild.rs/zh/config/tools/rspack#function-%E7%B1%BB%E5%9E%8B

```ts
import { Application } from 'typedoc';

pluginTypeDoc({
  typedoc: async () => {
    return Application.bootstrapWithPlugins({
        name: config.title,
        entryPoints,
        disableSources: true,
        router: 'kind',
        readme: 'none',
        githubPages: false,
        requiredToBeDocumented: ['Class', 'Function', 'Interface'],
        // @ts-expect-error - Typedoc does not export a type for this options
        plugin: [loadPluginMarkdown],
        entryFileName: 'index',
        hidePageHeader: true,
        hideBreadcrumbs: true,
        pageTitleTemplates: {
          module: '{kind}: {name}', // e.g. "Module: MyModule"
        },
      });
  }
})
```